tests/ga/extension_emulator.py

```diff
@@ -114,6 +113,8 @@ def generate_put_handler(*emulators):
     For use with tests.protocol.mocks.mock_wire_protocol.
     """
 
+    first_matching_emulator = lambda matches_func: next(emulator for emulator in emulators if matches_func(emulator))
```

Comment:
I see multiple copies of this helper function. Why not make it a private function of the `ExtensionEmulator` class and call that from everywhere. The uber goal is to avoid code duplication as much as possible